incuna-sass/mixins/_em-scale.sass

```diff
@@ -23,30 +26,14 @@
 
 // You can also use variables to reference media query variables in your
 // project:
-//     @include em-scale(14px, $px768, 16px, $px1025, 20px)
+//     @include em-scale(14px, (($px768, 16px), ($px1025, 20px)))
 // or
-//     @include em-scale(14px, $tablet, 16px, $desktop, 20px)
+//     @include em-scale(14px, (($tablet, 16px), ($desktop, 20px)))
 
-
-@mixin em-scale($font-size-1, $breakpoint-1, $font-size-2, $breakpoint-2: false, $font-size-3: false, $breakpoint-3: false, $font-size-4: false, $breakpoint-4: false, $font-size-5: false, $breakpoint-5: false, $font-size-6: false)
+@mixin em-scale($font-base, $breakpoints-sizes-list: ())
     font:
-        size: $font-size-1
-    @include media($breakpoint-1)
-        font:
-            size: $font-size-2
-    @if $breakpoint-2
-        @include media($breakpoint-2)
-            font:
-                size: $font-size-3
-    @if $breakpoint-3
-        @include media($breakpoint-3)
-            font:
-                size: $font-size-4
-    @if $breakpoint-4
-        @include media($breakpoint-2)
-            font:
-                size: $font-size-3
-    @if $breakpoint-5
-        @include media($breakpoint-3)
+        size: $font-base
+    @each $change in $breakpoints-sizes-list
+        @include media(#{nth($change, 1)})
             font:
-                size: $font-size-4
+                size: nth($change, 2)
```

Comment:
`$list-item` is already a name for each item in the mixin argument, where each item is a list of two items: breakpoint, font-size.

As an example of the nested structure (not literal code):
```sass
$breakpoints-sizes-list: ((a, 1), (b, 2))
...
$list-item-1: (a, 1)
$list-item-2: (b, 2)
...
$breakpoint-1: a
$font-size-1: 1
```